### PR TITLE
Feature/icon

### PR DESCRIPTION
**Pull request by OlegSymonchuk**
[Trello task](https://trello.com/c/8fgGYpzF)
[Visual design](https://www.figma.com/file/0KvoxIcY6iFSdVqZhwpOEd/MathRandom_web?node-id=14%3A0)

Changes:

-  add icon component
-  add demo page

### CODE REVIEW

**Review by ShevtsovAnton — APPROVED** [review]

**Review by kucherenko — APPROVED** [review]

File: pages/demo/blocks.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import styled from 'styled-components';
```
Comment (machulan) [inline pages/demo/blocks.tsx]: It would be better off deleting the file and moving the demonstration icons to storybook (Icon.stories.tsx).

**Review by machulan — COMMENTED** [review]
I suggest not using package-lock.json in order to avoid possible problems. Maintaining actual yarn.lock is enough. Please, use yarn only.

File: components/common/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+`;
+
+interface ImgComponentProps {
+  className: string;
+}
+
+const getImgComponent = (iconType: IconTypes): React.ComponentType<ImgComponentProps> =>
+  dynamic(() => import(`./inlineImages/${iconType}`), {ssr: true});
+
+function Icon({iconType, className, text, width, height, size = IconSizes.Medium, isActive = false, isTextOnLeft = false}: IconProps) {
```
Comment (machulan) [inline components/common/Icon/Icon.tsx]: The Icon is responsible not only for an icon but also for text. I guess text is not a part of an icon.

File: components/common/Icon/inlineImages/AttachFile.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+type PropsType = {
```
Comment (machulan) [inline components/common/Icon/inlineImages/AttachFile.tsx]: The PropsType can be reused by all the icons in /inlineImages

File: utils/svg_to_tsx.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import fs from 'fs';
```
Comment (machulan) [inline utils/svg_to_tsx.ts]: Using an svg library would be better for maintainability

File: styles/globalStyle.ts [hunk 5]
```diff
@@ -57,7 +65,7 @@ const fontStyle = css`
 `;
 
 const bodyStyle = css`
-  ${fontStyle}
+  ${fontStyle};
```
Comment (machulan) [inline styles/globalStyle.ts]: The semicolon looks useless.

Comment (OlegSymonchuk) [reply styles/globalStyle.ts]: It was added as ESLint highlighted the next line. Removed.

File: pages/demo/blocks.tsx (hunk 1 above)
Comment (OlegSymonchuk) [reply pages/demo/blocks.tsx]: It was in the storybook (Icon.stories.tsx). Demo page has been removed.

File: components/common/Icon/Icon.tsx (hunk 2 above)
Comment (OlegSymonchuk) [reply components/common/Icon/Icon.tsx]: Removed text

File: components/common/Icon/inlineImages/AttachFile.tsx (hunk 3 above)
Comment (OlegSymonchuk) [reply components/common/Icon/inlineImages/AttachFile.tsx]: Fixed by moving props into interfaces file

File: utils/svg_to_tsx.ts (hunk 4 above)
Comment (OlegSymonchuk) [reply utils/svg_to_tsx.ts]: Will be refactored later.

**Review by kucherenko — APPROVED** [review]

**Pull request merged**